Code/Engine/RendererCore/Meshes/Implementation/SkinnedMeshComponent.cpp

```diff
@@ -54,30 +52,10 @@ void ezSkinningState::TransformsChanged()
     m_hGpuBuffer = ezGALDevice::GetDefaultDevice()->CreateBuffer(BufferDesc, m_Transforms.GetArrayPtr().ToByteArray());
 
     ezGALDevice::GetDefaultDevice()->GetBuffer(m_hGpuBuffer)->SetDebugName("ezSkinningState");
-
-    m_bTransformsUpdated[0] = std::make_shared<bool>(true);
-    m_bTransformsUpdated[1] = std::make_shared<bool>(true);
   }
   else
   {
-    const ezUInt32 uiRenIdx = ezRenderWorld::GetDataIndexForExtraction();
-    *m_bTransformsUpdated[uiRenIdx] = false;
-  }
-}
-
-void ezSkinningState::FillSkinnedMeshRenderData(ezSkinnedMeshRenderData& ref_renderData) const
-{
-  ref_renderData.m_hSkinningTransforms = m_hGpuBuffer;
-
-  const ezUInt32 uiExIdx = ezRenderWorld::GetDataIndexForExtraction();
-
-  if (m_bTransformsUpdated[uiExIdx] && *m_bTransformsUpdated[uiExIdx] == false)
-  {
-    auto pSkinningMatrices = EZ_NEW_ARRAY(ezFrameAllocator::GetCurrentAllocator(), ezShaderTransform, m_Transforms.GetCount());
-    pSkinningMatrices.CopyFrom(m_Transforms);
-
-    ref_renderData.m_pNewSkinningTransformData = pSkinningMatrices.ToByteArray();
-    ref_renderData.m_bTransformsUpdated = m_bTransformsUpdated[uiExIdx];
+    ezGALDevice::GetDefaultDevice()->UpdateBufferForNextFrame(m_hGpuBuffer, m_Transforms.GetByteArrayPtr());
```

Comment:
Does this introduce a frame delay ?
Generally, does this affect performance in any way?

Reply:
No it does not introduce a frame delay and it allows us to have even better performance in the future than before. Although atm the performance is exactly the same.